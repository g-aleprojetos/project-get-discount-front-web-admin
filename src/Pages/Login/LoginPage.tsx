import React, {useCallback} from 'react';
import {Button} from 'Components/button';
import {Input} from 'Components/input';
import * as S from './LoginPage.styles';

export const LoginPage = () => {
  const [login, setLogin] = React.useState({email: '', senha: ''});

  const handleSubmit = useCallback(
    (event: React.FormEvent<HTMLFormElement>) => {
      event.preventDefault();
      return login;
    },
    [login],
  );

  const handleEsqueceuSenha = useCallback(() => {
    return 'recuperar senha';
  }, []);

  return (
    <S.Main data-testid={'loginPage'}>
      <S.Form
        autoComplete="off"
        data-testid={'loginPage-form'}
        onSubmit={handleSubmit}>
        <S.TextoTitulo>Login</S.TextoTitulo>
        <Input
          id="email"
          testId="input-email"
          label="Seu email"
          placeholder="contato@email.com"
          tipo="email"
          value={login.email}
          handleOnchange={(event: React.ChangeEvent<HTMLInputElement>) =>
            setLogin({...login, email: event.target.value})
          }
        />
        <Input
          id="senha"
          testId="input-senha"
          label="Sua senha"
          placeholder="*********"
          tipo="password"
          value={login.senha}
          handleOnchange={event =>
            setLogin({...login, senha: event.target.value})
          }
        />
        <Button testId="bnt-login" tipo="submit">
          <S.TextoBotao>Logar</S.TextoBotao>
        </Button>
        <S.BotaoRecuperaSenha
          data-testid={'botao-recuperar-senha'}
          onClick={handleEsqueceuSenha}>
          <S.TextoRecuperarSenha>Esqueceu sua senha?</S.TextoRecuperarSenha>
        </S.BotaoRecuperaSenha>
      </S.Form>
    </S.Main>
  );
};
