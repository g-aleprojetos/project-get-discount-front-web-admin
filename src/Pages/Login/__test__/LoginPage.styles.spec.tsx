import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../LoginPage.styles';

describe('LoginPage.styles', () => {
  test('BotaoRecuperaSenha DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.BotaoRecuperaSenha />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Form DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Form />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Main DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Main />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('TextoBotao DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.TextoBotao />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('TextoRecuperarSenha DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.TextoRecuperarSenha />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('TextoTitulo DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.TextoTitulo />).toJSON();

    expect(container).toMatchSnapshot();
  });
});
