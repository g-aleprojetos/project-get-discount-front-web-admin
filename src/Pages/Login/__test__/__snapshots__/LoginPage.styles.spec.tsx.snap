// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage.styles BotaoEnviar DEVE ser igual ao snapshot 1`] = `
.c0 {
  margin: 15px;
}

.c1 {
  min-width: 200px;
  background-color: #6495ED;
  border-radius: 8px;
  border: none;
  padding: 8px 60px;
  cursor: pointer;
}

<div
  className="c0"
  data-testid="bnt-pardao"
>
  <button
    className="c1"
    data-testid="bnt-pardao-button"
    type="submit"
  />
</div>
`;

exports[`LoginPage.styles ContainerBotaoRecuperaSenha DEVE ser igual ao snapshot 1`] = `
.c0 {
  background-color: transparent;
  border: none;
  cursor: pointer;
}

<button
  className="c0"
/>
`;

exports[`LoginPage.styles ContainerDireita DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 50%;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;

exports[`LoginPage.styles ContainerEsquerda DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 50%;
  height: 100%;
  background-color: #0000CD;
}

<div
  className="c0"
/>
`;

exports[`LoginPage.styles ContainerRecuperarSenha DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 100%;
  padding-right: 8px;
}

<div
  className="c0"
/>
`;

exports[`LoginPage.styles Form DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 400px;
  padding: 20px;
  gap: 8px;
  border-color: #000000;
}

<form
  className="c0"
/>
`;

exports[`LoginPage.styles Main DEVE ser igual ao snapshot 1`] = `
.c0 {
  width: 100vw;
  height: 100vh;
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #D3D3d3;
}

<main
  className="c0"
/>
`;

exports[`LoginPage.styles TextoBotao DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 18px;
  font-family: AvertaStd-Regular;
  color: #FFFFFF;
  margin-top: 0px;
  text-transform: none;
  line-height: 26px;
  cursor: pointer;
}

<h2
  className="c0 "
  cursor="pointer"
/>
`;

exports[`LoginPage.styles TextoRecuperarSenha DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 16px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: 24px;
  cursor: pointer;
}

<h3
  className="c0 "
  cursor="pointer"
/>
`;

exports[`LoginPage.styles TextoTitulo DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-family: AvertaStd-Bold;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: 34px;
  cursor: text;
}

<h1
  className="c0 "
/>
`;
